### PR TITLE
Add com.umlet.Umlet

### PR DESCRIPTION
**Pull request by zocker-160**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

`filesystem=home` is needed in order to make the Java AWT file picker work, which sadly does not support the portal (yet)
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/umlet/umlet/issues/640

Upstream developers are currently not interested in Flatpak, but one contributor mentioned, that they might potentially support it officially in the future.
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).

I do not own the domain, but I got permission by upstream developers to submit this package to FlatHub.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.umlet.Umlet.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        - /usr/lib/sdk/openjdk11/install.sh
+
+    - name: Umlet
+      buildsystem: simple
+      build-commands:
+        - install -D -m755 umlet.jar -t /app/lib/umlet
+        - cp -a {lib,palettes,custom_elements} /app/lib/umlet
+        - install -D -m755 umlet.sh /app/bin/umlet
+        # fix start command
+        - sed -i -e 's/Exec=java -Dsun.java2d.xrender=f -jar umlet.jar/Exec=umlet/g' umlet.desktop
```
Comment (hfiguiere) [inline com.umlet.Umlet.yml:35]: you should use `desktop-file-edit`

File: com.umlet.Umlet.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        - install -D -m644 umlet.desktop -t /app/share/applications
+        - install -D -m644 umlet_logo.png -t /app/share/icons/hicolor/512x512/apps
+      post-install:
```
Comment (hfiguiere) [inline com.umlet.Umlet.yml:38]:
```suggestion
```

File: com.umlet.Umlet.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+rename-icon: umlet_logo
+rename-desktop-file: umlet.desktop
+rename-appdata-file: metainfo.xml
+command: umlet
+
+finish-args:
+    - --share=ipc
+    - --socket=x11
+    - --device=dri
+    - --filesystem=home
```
Comment (hfiguiere) [inline com.umlet.Umlet.yml:17]: you could limit to xdg-documents

File: com.umlet.Umlet.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    - --filesystem=xdg-config/fontconfig:ro
```
Comment (hfiguiere) [inline com.umlet.Umlet.yml:18]: why is this needed?

File: com.umlet.Umlet.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        - install -D -m644 umlet.desktop -t /app/share/applications
+        - install -D -m644 umlet_logo.png -t /app/share/icons/hicolor/512x512/apps
+      post-install:
+        - install -D -m644 metainfo.xml -t /app/share/metainfo
+      sources:
+        - type: archive
+          url: https://www.umlet.com/download/umlet_15_0/umlet-standalone-15.0.0.zip
+          sha256: 81dbe1a981b2ac5b90861ae4176eb05dbdd340b4422e6e7dfee4b14cf9877401
+        - type: file
+          url: https://raw.githubusercontent.com/umlet/umlet/0da4d7d175fd52283ed3a5e8fe38d4723e7ac51f/umlet-standalone/src/exe/img/umlet_logo.png
```
Comment (hfiguiere) [inline com.umlet.Umlet.yml:45]: this image is in the zip. there is no point downloading it separately.

**Comment by hfiguiere** [comment]
bot, build com.umlet.Umlet

File: com.umlet.Umlet.yml (hunk 1 above)
Comment (zocker-160) [reply com.umlet.Umlet.yml]: fixed

File: com.umlet.Umlet.yml (hunk 2 above)
Comment (zocker-160) [reply com.umlet.Umlet.yml]: done

File: com.umlet.Umlet.yml (hunk 3 above)
Comment (zocker-160) [reply com.umlet.Umlet.yml]: done

File: com.umlet.Umlet.yml (hunk 5 above)
Comment (zocker-160) [reply com.umlet.Umlet.yml]: fixed

File: com.umlet.Umlet.yml (hunk 4 above)
Comment (zocker-160) [reply com.umlet.Umlet.yml:18]: this is a bit tricky, this is an attempt to make the font not look absolutely terrible, also see https://github.com/flatpak/flatpak/issues/4122

Adding that line, makes the font at least aliased, which is better than nothing I guess.

**Comment by zocker-160** [comment]
bot, build com.umlet.Umlet

File: com.umlet.Umlet.yml (hunk 5 above)
Comment (zocker-160) [reply com.umlet.Umlet.yml:45]: ok I had to revert this, because the icon in the zip is not squared, because the commit which fixes this problem happened after the release of the version.

So I have to download it separately until upstream published a new version which contains that fix.

**Comment by zocker-160** [comment]
bot, build com.umlet.Umlet

File: com.umlet.Umlet.yml (hunk 5 above)
Comment (hfiguiere) [reply com.umlet.Umlet.yml:45]: ah ok.
